art_internal_impl.hpp

```diff
@@ -1370,22 +1724,93 @@ class basic_inode_16 : public basic_inode_16_parent<ArtPolicy> {
   }
   UNODB_DETAIL_RESTORE_MSVC_WARNINGS()
 
-  constexpr void delete_subtree(db &db_instance) noexcept {
+  [[nodiscard]] constexpr node_ptr get_child(
+      const std::uint8_t child_index) noexcept {
+    return children[child_index].load();
+  }
+
+  // N16 - position on the first child.
+  [[nodiscard]] constexpr typename basic_inode_16::iter_result
+  begin() noexcept {
+    const auto key = keys.byte_array[0].load();
+    return {node_ptr{this, node_type::I16}, key, 0};
+  }
+
+  // N16 - position on the last child.
+  [[nodiscard]] constexpr typename basic_inode_16::iter_result last() noexcept {
+    const std::uint8_t child_index = this->children_count.load() - 1;
+    const auto key = keys.byte_array[child_index].load();
+    return {node_ptr{this, node_type::I16}, key, child_index};
+  }
+
+  [[nodiscard]] constexpr typename basic_inode_16::iter_result_opt next(
+      const std::uint8_t child_index) noexcept {
+    const std::uint8_t nchildren = this->children_count.load();
+    const std::uint8_t next_index = child_index + 1;  // next child index
+    if (next_index >= nchildren) return parent_class::end_result;
+    const auto key = keys.byte_array[next_index].load();
+    return {{node_ptr{this, node_type::I16}, key, next_index}};
+  }
+
+  [[nodiscard]] constexpr typename basic_inode_16::iter_result_opt prior(
+      const std::uint8_t child_index) noexcept {
+    if (child_index == 0) return parent_class::end_result;
+    const std::uint8_t next_index = child_index - 1;  // next child index
+    const auto key = keys.byte_array[next_index].load();
+    return {{node_ptr{this, node_type::I16}, key, next_index}};
+  }
+
+  // N16: The keys[] is ordered, so we scan the keys[] in reverse
+  // order, returning the first value LTE the given [key_byte].
+  [[nodiscard]] constexpr typename basic_inode_16::iter_result_opt lte_key_byte(
+      std::byte key_byte) noexcept {
+    const std::uint8_t children_count_ = this->children_count.load();
```

Comment:
Add a TODO to look into SIMD implementation

Reply:
One note to rule them all